src/ctia/flows/crud.clj

```diff
@@ -0,0 +1,75 @@
+(ns ctia.flows.crud
+  "This namespace handle all necessary flows for creating, updating and deleting objects.
+
+  (Cf. #159)."
+  (:import java.util.UUID)
+  (:require [ctia.flows.hooks :refer :all]))
+
+(defn make-id
+  "Apprently `make-id` is the same for all stores."
+  [type-name _]
+  (str (name type-name) "-" (UUID/randomUUID)))
+
+(defn create-flow
+  "This function centralize the create workflow.
+  It is helpful to easily add new hooks name
+
+  To be noted:
+    - `:before-create` hooks can modify the object stored.
+    - `:after-create` hooks can't modify the object returned but
+      a hook modifying an object will provide the modified object
+      to the next `:after-create` hook and so on."
+  [realize-fn
```

Comment:
Please consider using maps and destructuring for these fns.  Its hard to keep track of the args when reading the handler code.